lib/GRNOC/TSDS/Setup.pm

```diff
@@ -0,0 +1,887 @@
+package GRNOC::TSDS::Setup;
+
+use Moo;
+
+use GRNOC::TSDS;
+use GRNOC::CLI;
+use GRNOC::Config;
+
+use GRNOC::TSDS::Install;
+
+use Data::Dumper;
+use MongoDB;
+use Sort::Versions;
+use Sys::Hostname;
+### required attributes ###
+
+has config_file => ( is => 'ro',
+    required => 1 );
+
+has unattended => ( is => 'ro',
+    required => 1 );
+
+### internal attributes ###
+
+has cli => ( is => 'ro',
+    default => sub { return GRNOC::CLI->new() } );
+
+has config => ( is => 'rwp' );
+
+has mongo_root => ( is => 'rwp' );
+
+has error => ( is => 'rwp' );
+
+### constructor builder ###
+
+sub BUILD {
+
+    my ( $self ) = @_;
+
+    my $config = GRNOC::Config->new( config_file => $self->config_file,
+        force_array => 0 );
+
+    $self->_set_config( $config );
+
+}
+
+### public methods ###
+
+sub upgrade{
```

Comment:
Related to earlier comment - just change this method name. We're not doing upgrades, we're doing a setup here.

Reply:
Changed to `setup`.